### PR TITLE
sys-sage: change the target repo and update the version number with a…

### PR DESCRIPTION
**Pull request by stepanvanecek**
…dded options

### CODE REVIEW

File: var/spack/repos/builtin/packages/sys-sage/package.py [hunk 1]
```diff
@@ -3,23 +3,79 @@
...
 
-    homepage = "https://github.com/stepanvanecek/sys-sage"
-    url = "https://github.com/stepanvanecek/sys-sage/archive/refs/tags/v0.1.1-alpha.2.tar.gz"
-    git = "https://github.com/stepanvanecek/sys-sage.git"
+    homepage = "https://github.com/caps-tum/sys-sage"
+    url = "https://github.com/caps-tum/sys-sage/archive/refs/tags/v0.4.3.tar.gz"
+    git = "https://github.com/caps-tum/sys-sage.git"
 
-    maintainers("stepanvanecek")
+    maintainers = ["stepanvanecek"]
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/sys-sage/package.py]: Please revert this change, `maintainers` has been a directive for a while:
```suggestion
    maintainers("stepanvanecek")
```

File: var/spack/repos/builtin/packages/sys-sage/package.py [hunk 2]
```diff
@@ -3,23 +3,79 @@
...
+    depends_on("hwloc@2.9:", when="+build_data_sources")
+    depends_on("hwloc@2.9:", when="+ds_hwloc")
+    depends_on("cuda", when="+nvidia_mig platform=linux")
+    depends_on("cuda", when="+build_data_sources platform=linux")
+
+    def cmake_args(self):
+        spec = self.spec
+        args = []
+        if "+nvidia_mig" in spec:
+            args.append("-DNVIDIA_MIG=ON")
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/sys-sage/package.py]: You can use `self.define` and `self.define_from_variant` to make the recipe more legible.

**Review by alalazo — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/sys-sage/package.py (hunk 2 above)
Comment (stepanvanecek) [reply var/spack/repos/builtin/packages/sys-sage/package.py]: Thanks for the comments, I updated the recipe.

File: var/spack/repos/builtin/packages/sys-sage/package.py [hunk 3]
```diff
@@ -3,23 +3,78 @@
...
+    variant(
+        "cpuinfo",
+        default=True,
+        description="Build and install functionality regarding Linux cpuinfo (only x86) -- "
+        "default ON.",
+    )
+    variant(
+        "build_data_sources",
+        default=False,
+        description="Build all data sources (programs to collect data about the machine sys-sage "
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/sys-sage/package.py]:
```suggestion
        default=False,
        when="platform=linux",
        description="Build all data sources (programs to collect data about the machine sys-sage "
```

File: var/spack/repos/builtin/packages/sys-sage/package.py [hunk 4]
```diff
@@ -3,23 +3,78 @@
...
+    )
+    variant(
+        "ds_hwloc",
+        default=False,
+        description="Builds the hwloc data source for retrieving the CPU topology",
+    )
+    variant(
+        "ds_numa",
+        default=False,
+        description="builds the caps-numa-benchmark. If turned on, includes Linux-specific "
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/sys-sage/package.py]:
```suggestion
        default=False,
        when="platform=linux",
        description="builds the caps-numa-benchmark. If turned on, includes Linux-specific "
```

File: var/spack/repos/builtin/packages/sys-sage/package.py [hunk 5]
```diff
@@ -3,23 +3,78 @@
...
+        else:
+            args.append(self.define("CPUINFO", False))
+        if "+ds_hwloc" in spec or "+build_data_sources" in spec:
+            args.append(self.define("DS_HWLOC", True))
+        if "+ds_numa" in spec or "+build_data_sources" in spec:
+            if spec.platform == "linux":
+                args.append(self.define("DS_NUMA", True))
+            else:
+                msg = "ds_numa option is only available on Linux systems."
+                raise InvalidSpecDetected(msg)
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/sys-sage/package.py]: You should be able to make the spec fail at concretization time, rather than installation time:
```suggestion
            args.append(self.define("DS_NUMA", True))
```

**Review by alalazo — CHANGES_REQUESTED** [review]
I left a suggestion. Let me know if it works.

File: var/spack/repos/builtin/packages/sys-sage/package.py (hunk 5 above)
Comment (stepanvanecek) [reply var/spack/repos/builtin/packages/sys-sage/package.py]: Thanks for the suggestions -- this way, the recipe is also easier to read.

**Comment by stepanvanecek** [comment]
@alalazo I accepted the changes you suggested, thanks for the suggestions. Does the recipe look good to you now?

**Review by alalazo — APPROVED** [review]
